packages/plugin-flag/test/integration/spec/flag.js

```diff
@@ -0,0 +1,180 @@
+        conversation.activities.items.shift(); // shift to get rid of the create activity at the top
+        const comments = map(conversation.activities.items, 'object.displayName');
+        assert.lengthOf(comments, 4);
+        assert.equal(comments[0], 'Hi Dear, How are you?');
+        assert.equal(comments[1], 'Hey! I am doing well. How are you?');
+        assert.equal(comments[2], 'I am also doing well. Are you in for the party?');
+        assert.equal(comments[3], 'Yes, I am in.');
+        flagConversation = conversation;
+      })
+    );
+
+
+    afterEach(() => {
+      return spock.spark.flag.list()
+        .then((flags) => {
+          flags.forEach((flag) => {
+            return spock.spark.flag.remove(flag);
+          });
+        });
+    });
+
+    describe('#flag()', () => {
+      it('flags the activity', () => {
+        assert(true);
+        const flaggedActivity1 =  flagConversation.activities.items[1];
+        return spock.spark.flag.flag(flaggedActivity1)
+          .then((flagResponse1) => {
+            assert.equal(flagResponse1.state, 'flagged');
+          });
+      });
+    });
+
+    describe('#list()', () => {
+      it('fetches the flag list', () => {
+        return spock.spark.flag.list()
+          .then((flagList) => {
+            assert.isArray(flagList);
+            assert.lengthOf(flagList, 0);
+          });
+      });
+    });
+
+    describe.only('#mapToActivities()', () => {
+      it('maps flags to activity', () => {
+        const flaggedActivity1 =  flagConversation.activities.items[1];
+        return spock.spark.flag.flag(flaggedActivity1)
+          .then((flagResponse1) => {
+            assert.equal(flagResponse1.state, 'flagged');
+            let flags = [];
+            flags.push(flagResponse1);
+            return spock.spark.flag.mapToActivities(flags)
+              .then((activities) => {
+                const activity = activities[0];
+                assert.equal(activity.object.displayName, 'Hey! I am doing well. How are you?');
+                assert.isDefined(find(activities, {url: flagResponse1['flag-item']}));
+              });
+          });
+      });
+    });
+
+    describe('#remove()', () => {
+      it('removes the flag from activity', () => {
+        const flaggedActivity1 =  flagConversation.activities.items[1];
+        return spock.spark.flag.flag(flaggedActivity1)
+          .then((flagResponse1) => {
+            assert.equal(flagResponse1.state, 'flagged');
+            return spock.spark.flag.remove(flagResponse1);
+          })
+          .then(() => spock.spark.flag.list())
+          .then((flagList) => {
+            assert.isArray(flagList);
+            assert.lengthOf(flagList, 0);
+          });
+      });
+    });
+
+    describe('#archive()', () => {
+      it('archives the flag for an activity', () => {
+        const flaggedActivity1 =  flagConversation.activities.items[1];
+        return spock.spark.flag.flag(flaggedActivity1)
```

Comment:
Would it make sense to change the methods names to 

`create`, `get`, `list`, `delete`, `update`, etc to avoid cases like `flag.flag`?